app/models/entrant.rb

```diff
@@ -0,0 +1,33 @@
+class Entrant < ApplicationRecord
+  belongs_to :raceable, polymorphic: true, touch: true
+  belongs_to :user
+
+  def ready?
+    readied_at.present?
+  end
+
+  def finished?
+    finished_at.present?
+  end
+
+  def forfeited?
+    forfeited_at.present?
+  end
+
+  def done?
+    finished? || forfeited?
+  end
+
+  def place
+    return '-' unless finished?
+    return 'big fat x' if forfeited?
+
+    raceable.entrants.order(finished_at: :asc).pluck(:id).index(id) + 1
+  end
+
+  def duration
+    return nil unless finished?
```

Comment:
In the past week or so I've merged some commits that apply a small twist on the [null object pattern][1] to Duration -- if you use a `Duration.new(nil)` rather than `nil`, it will mostly behave smartly when it's compared / rendered / used in arithmetic with other Durations.

[1]: https://medium.com/@franzejr/null-object-pattern-in-ruby-12a809bc75cc

Reply:
I think I had half of those changes when I branched off, but having that built in support for `nil` will be much nicer to work with.